jdaviz/configs/default/plugins/data_tools/data_tools.py

```diff
@@ -1,54 +1,67 @@
     error_message = Unicode().tag(sync=True)
+    directory = Unicode(None, allow_none=True).tag(sync=True)
+    file = Unicode(None, allow_none=True).tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         start_path = os.environ.get('JDAVIZ_START_DIR', os.path.curdir)
+        start_path = str(Path(start_path).resolve())
 
-        self._file_upload = FileChooser(start_path)
+        def on_directory_change(directory : Path):
+            self.directory = str(directory)
 
-        self.components = {'g-file-import': self._file_upload}
+        def on_file_open(file : Path):
+            self._load(file)
+
+        def on_path_select(path : Optional[Path]):
+            if path is None:
+                self.directory = None
+                self.file = None
+            else:
+                if path.is_dir():
+                    self.directory = str(path)
+                else:
+                    self.file = str(path)
 
-        self._file_upload.observe(self._on_file_path_changed, names='file_path')
+        element = sol.FileBrowser(start_path,
+                                  on_directory_change=on_directory_change,
+                                  on_file_open=on_file_open,
+                                  on_path_select=on_path_select,
+                                  can_select=True,
+                                  )
+        self._file_upload, rc = react.render(element)
 
-    def _on_file_path_changed(self, event):
-        if (self._file_upload.file_path is not None
-                and not os.path.exists(self._file_upload.file_path)
-                or not os.path.isfile(self._file_upload.file_path)):
-            self.error_message = "No file exists at given path"
-            self.valid_path = False
+        self.components = {'g-file-import': self._file_upload}
+
+    def _load(self, path : Path):
+        try:
+            load_data_message = LoadDataMessage(str(path), sender=self)
+            self.hub.broadcast(load_data_message)
+        except Exception:
+            self.error_message = "An error occurred when loading the file"
         else:
-            self.error_message = ""
-            self.valid_path = True
-
-    def vue_load_data(self, *args, **kwargs):
-        if self._file_upload.file_path is None:
-            self.error_message = "No file selected"
-        elif os.path.exists(self._file_upload.file_path):
-            try:
-                load_data_message = LoadDataMessage(self._file_upload.file_path, sender=self)
-                self.hub.broadcast(load_data_message)
-            except Exception as err:
-                self.error_message = f"An error occurred when loading the file: {repr(err)}"
-            else:
-                self.dialog = False
+            self.dialog = False
+
+    def vue_load_file(self, *args, **kwargs):
+        self._load(Path(self.file))
 
-            if self.config == 'imviz':
```

Comment:
We will still need this no matter which solution you are going with.

Reply:
Thanks, I missed that during rebasing.